### PR TITLE
Support tuples in T::Struct props

### LINKED ISSUE #1991: T::Struct props silently dropped if they contain tuples
#### Input

[Sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%0Aclass%20Thing%3B%20end%0Aclass%20Thing2%3B%20end%0A%0Aclass%20A%20<%20T%3A%3AStruct%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20const%20%3Aarr_things%2C%20T%3A%3AArray%5BThing%5D%0A%20%20const%20%3Aarr_arr_things%2C%20T%3A%3AArray%5BT%3A%3AArray%5BThing%5D%5D%0A%20%20const%20%3Atuples%2C%20%5BThing%2C%20Thing2%5D%0A%20%20const%20%3Aarr_tuples%2C%20T%3A%3AArray%5B%5BThing%2C%20Thing2%5D%5D%20%20%0Aend%0A%0Adef%20main%0A%20%20A.new(%0A%20%20%20%20arr_things%3A%20%5B%5D%2C%20%23%20OK%0A%20%20%20%20arr_arr_things%3A%20%5B%5D%2C%20%23%20OK%0A%20%20%20%20tuples%3A%20%5BThing.new%2C%20Thing2.new%5D%2C%20%23%20Fails%0A%20%20%20%20arr_tuples%3A%20%5B%5D%2C%20%23%20Fails%0A%20%20)%0Aend)

```ruby
# typed: true

class Thing; end
class Thing2; end

class A < T::Struct
  extend T::Sig

  const :arr_things, T::Array[Thing]
  const :arr_arr_things, T::Array[T::Array[Thing]]
  const :tuples, [Thing, Thing2]
  const :arr_tuples, T::Array[[Thing, Thing2]]  
end

def main
  A.new(
    arr_things: [], # OK
    arr_arr_things: [], # OK
    tuples: [Thing.new, Thing2.new], # Fails
    arr_tuples: [], # Fails
  )
end
```

#### Observed output

```
editor.rb:16: Unrecognized keyword argument tuples passed for method A#initialize https://srb.help/7004
    16 |  A.new(
    17 |    arr_things: [], # OK
    18 |    arr_arr_things: [], # OK
    19 |    tuples: [Thing.new, Thing2.new], # Fails
    20 |    arr_tuples: [], # Fails
    21 |  )

editor.rb:16: Unrecognized keyword argument arr_tuples passed for method A#initialize https://srb.help/7004
    16 |  A.new(
    17 |    arr_things: [], # OK
    18 |    arr_arr_things: [], # OK
    19 |    tuples: [Thing.new, Thing2.new], # Fails
    20 |    arr_tuples: [], # Fails
    21 |  )
Errors: 2
```

#### Expected behavior

Sorbet DSL pass should error when it comes across tuples in prop definition, as opposed to silently dropping them and then erroring at the site where they're referenced.ee

- - -

### PR DESCRIPTION
**Pull request by jez**
### Motivation

Fixes #1991.

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by jez** [comment]
We have a policy of testing changes to Sorbet against Stripe's codebase before
merging them. I've kicked off a test run for the current PR.  When the build
finishes, I'll share with you whether or how it failed. Thanks!

Stripe employees can see the build results here:

→ https://go/builds/bui_JqbOWAwlXDuvcB
→ https://go/builds/bui_JqbObAsmOlbxFr

File: test/testdata/rewriter/prop_tuple.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# typed: true
+extend T::Sig
+
+class StructWithTuples < T::Struct
+  prop :unary_tuple, [Symbol]
+  prop :tuple, [Integer, String]
+  prop :triple, [Float, TrueClass, FalseClass]
+  prop :array_of_tuple, T::Array[[Integer, String]]
+  prop :tuple_with_combinators, [T.nilable(String), T.class_of(Integer)]
+  prop :combinator_with_tuple, T.nilable([Symbol, Float])
```
Comment (aprocter-stripe) [inline test/testdata/rewriter/prop_tuple.rb:10]: OOC, do we already have tests for what happens if the types for `prop` declarations are not syntactically valid? (e.g., `prop :some_tuple, [42, 42, 42]`)

**Review by aprocter-stripe — APPROVED** [review]

File: test/testdata/rewriter/prop_tuple.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+end
```
Comment (jez) [inline test/testdata/rewriter/prop_tuple.rb:11]:
```suggestion
end

class StructWithBadTuples < T::Struct
  prop :bad_literal, [42, 42]
end
```

File: test/testdata/rewriter/prop_tuple.rb (hunk 1 above)
Comment (jez) [reply test/testdata/rewriter/prop_tuple.rb:10]: Added below.

**Pull request merged**